src/lib/workspace/Subsetting/SubsetDownloadModal.tsx

```diff
@@ -220,6 +225,14 @@ export default function SubsetDownloadModal({
     ({ pageSize, pageIndex, sortBy }) => {
       if (!currentEntity) return;
       setDataLoading(true);
+      setSortBy(sortBy ?? []);
+
+      const selectedVariableIds = selectedVariableDescriptorsWithMergeKeys.map(
+        (descriptor) => descriptor.variableId
+      );
+      const filteredSortBy = sortBy?.filter((column) =>
```

Comment:
am i reading this right that this line is basically a sanity check to ensure that everything we're about to sort by actually is in the variables we selected in the table? 
Was there a way to sort by a var that wasn't selected? Or was this just a cautionary add?

Reply:
Not quite a sanity check---I found a bug caused by the fact that `sortBy`'s state was lagging behind the currently selected variables when we removed a column with the X button. This meant in our request we were asking the backend to sort by a column that we didn't ask it to give us, so it was understandably confused about this and gave us a 500 instead. :)

I didn't see a good way to fix the inconsistency due to the logic flow, but paring down `sortBy` to have only selected variables works just as well.